### PR TITLE
fix: remove duplicated cell when interleave filter is applied

### PR DESCRIPTION
**Pull request by rahulKQL**
It seems Bigtable classic client & veneer client behaves differently for duplicated `cells`. A Row can contain duplicated `cells` when the user applies `Interleave` filters. The behavior of both of these clients are:
   - Classic client: Here, we check if a `cell` contains some `labels` or not, If it does then we include that cell in end result if it doesn't then we compare `timestamp` and `qualifier` with previous no label cell(Because a cell with a `label` could have been produced by applying filters).
   - Veneer client: Here, we do not perform these checks. (Most of the other client allows duplicate cells confirmed on `Go/NodeJs/C#` bigtable client. Not sure why but `python-bigtable` does not allow duplicate cells)

**Assumption:** The labels are applied to determine which filters produced those cells, So we are including all cells where labels are present.

### CODE REVIEW

**Review by pmakani — DISMISSED** [review]

**Review by athakor — DISMISSED** [review]

File: bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/RowResultAdapter.java [hunk 1]
```diff
@@ -159,6 +162,9 @@ public void cellValue(ByteString newValue) {
      * </ul>
      *
      * A flattened version of the {@link RowCell} map will be sorted correctly.
+     *
+     * <p>In case user applies {@link InterleaveFilter} than a {@link Cell} can appear more than
+     * once, but the duplicate cells will appear one after another.
      */
```
Comment (dmitry-fa) [inline bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/RowResultAdapter.java:178]: than -> then
it's not clear to me, what do you mean 'cells will appear one after another'

I would rephrase: 
<p>Applying {@link InterleaveFilter} may result in a row to contain duplicated cells.

Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/RowResultAdapter.java:178]: `cells will appear one after another` => I mean the duplicate cells will always be in group one after another (in that sequence)

Comment (dmitry-fa) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/RowResultAdapter.java:178]: no 'but' here.
may be:
> , but the duplicate cells will appear in a group i.e. one after another.

, where duplicates are grouped in sequences.

Comment (rahulKQL) [reply bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/RowResultAdapter.java:178]: Thanks for catching it 🙏

**Review by dmitry-fa — DISMISSED** [review]
LGTM
